layouts/_default/baseof.html

```diff
@@ -16,10 +16,26 @@
     {{ end }}
   >
     <div class="wrapper">
-      <aside class="wrapper__sidebar">
+      <aside
+        {{ if and (not .IsHome) (eq .Site.Params.hidesidebar true) }}
+          class="wrapper__sidebar--sidebarhidden"
+
+        {{ else }}
+          class="wrapper__sidebar"
+
+        {{ end }}
+      >
         {{- partial "sidebar.html" . -}}
       </aside>
-      <main class="wrapper__main">
+      <main
+        {{ if and (not .IsHome) (eq .Site.Params.hidesidebar true) }}
+          class="wrapper__main--sidebarhidden"
```

Comment:
Same concerns apply as described above. 

Great example here:

You add `width: 100%` when the sidebar is hidden:
```scss
@include desktop_and_print {
  width: $content-width;
  &--sidebarhidden {
    margin-top: 0px;
    width: 100%;
  }
}
```

However, `width: 100%` would already included in `wrapper__main`. So please make use of the extensible nature of BEM. That's what it's there for.

Reply:
Unfortunately, I believe in this case the width would be required as the desktop_and_print include that is applied to the wrapper__main would otherwise set the width to the content width. I am currently checking, if there is another way to make the code more lean.

Reply:
You're totally right. Sorry for missing that!